spec/models/user_spec.rb

```diff
@@ -53,5 +53,21 @@
         it { is_expected.to allow_value(email).for(:email) }
       end
     end
+
+    describe 'User roles' do
+      let(:university){create :user, email: 'oxford@oxford.edu', role: :university}
+    
```

Comment:
You need to test the `research_group` role as well.

Reply:
fixed